src/windows/exportClips.py

```diff
@@ -0,0 +1,236 @@
+        ui_util.init_ui
+        self.file_objs = export_clips_arg
+        self._createWidgets()
+
+    def setPath(self, p: str):
+        self.fp.setPath(p)
+
+    def _createWidgets(self):
+        self.fp = filePicker(folder_only=True)
+        self.export_button = QPushButton(_("Export"))
+        self.export_button.clicked.connect(self._exportPressed)
+        self.close_button = QPushButton(_("Close"))
+        self.close_button.clicked.connect(self.done)
+        self.cancel_button = QPushButton(_("Cancel"))
+        self.cancel_button.clicked.connect(self._cancelButtonClicked)
+
+        # Make progress bar look like the one in the export dialog
+        from PyQt5.QtGui import QPalette
+        p = QPalette()
+        p.setColor(QPalette.Highlight, Qt.green)
+        self.progressExportVideo.setPalette(p)
+
+        self.buttonBox.addButton(self.export_button, QDialogButtonBox.ActionRole)
+        self.buttonBox.addButton(self.cancel_button, QDialogButtonBox.ActionRole)
+        self.buttonBox.addButton(QDialogButtonBox.Close).clicked.connect(lambda: self.done(0))
+        self.FilePickerArea.addWidget(self.fp)
+        self.progressExportVideo.setValue(0)
+
+    def _cancelButtonClicked(self):
+        self.exporting = False
+
+    def _updateProgressBar(self, count: int, total: int):
+        d = count - total
+        if -2 <= d and 2 >= d:
+            self.progressExportVideo.setValue(100)
+            return
+        self.progressExportVideo.setValue((count/total) * 100)
+
+    def _exportPressed(self):
+        clips = list(filter(isClip, self.file_objs))
+        files = list(filter(notClip, self.file_objs))
+        # Total number of frames
+        self._updateDialogExportStarting()
+        export_destination = self.fp.getPath()
+        total_frames, frames_written = 0, 0
+        for c in clips:
+            total_frames += framesInClip(c)
+        for f in files:
+            total_frames += int(f.data.get("video_length",0))
+        for f in files:
+            copyFileToFolder(f, export_destination)
+            frames_written+=int(f.data.get("video_length",0))
+        for c in clips:
+            extension = "mp4"
+            export_path = os.path.join(export_destination, f"{nameOfExport(c)}.{extension}")
+            if(os.path.exists(export_path)):
+                log.info("Export path exists. Skipping render")
+                frames_written += framesInClip(c)
+                self._updateProgressBar(frames_written, total_frames)
+                get_app().processEvents()
+                continue
+            w = openshot.FFmpegWriter(export_path)
+            setupWriter(c, w)
+            w.PrepareStreams()
+            w.Open()
+
+            start_time = c.data.get("start")
+            end_time = c.data.get("end")
+            fps = c.data.get("fps").get("num") / c.data.get("fps").get("den")
+            start_frame, end_frame = int(start_time * fps), int(end_time*fps)
```

Comment:
Should this use round() instead of int()?

Reply:
In fact, you might move the "time to frame" calculation into a function...  The logic should be:

```
(time * fps) + 1   (to ensure we always start on frame 1)
```